src/parser.rs

```diff
@@ -1794,47 +1829,29 @@ impl Parser {
                     // Either we'll be able to parse a valid nested join, or
                     // we won't, and we'll return that error instead.
                     self.index = index;
-                    let table_and_joins = self.parse_table_and_joins()?;
-                    match table_and_joins.relation {
-                        TableFactor::NestedJoin { .. } => (),
-                        _ => {
-                            if table_and_joins.joins.is_empty() {
-                                // The SQL spec prohibits derived tables and bare
-                                // tables from appearing alone in parentheses.
-                                self.expected("joined table", self.peek_token())?
+                    match self.parse_table_inside_parenthesis() {
+                        Ok(res) => Ok(res), 
+                        Err(_) => {
+                            self.index = index;
+                            let table_and_joins = self.parse_table_and_joins()?; 
+                            match table_and_joins.relation {
+                                TableFactor::NestedJoin { .. } => (),
+                                _ => {
+                                    if table_and_joins.joins.is_empty() {
+                                        // The SQL spec prohibits derived tables and bare
```

Comment:
I expected the change to be here, as _"bare tables .. appearing alone in parentheses"_ is exactly the case you reported as supported in Snowflake SQL. Not sure about the "derived tables .. alone in parentheses" case.